app/views/curation_concern/permissions/confirm.html.erb

```diff
@@ -3,19 +3,11 @@
     <h4>Apply changes to contents?<h4>
   </div>
   <div class="panel-body">
-    <% if curation_concern.under_embargo? %>
-      <p>You've applied an embargo to this <%= curation_concern.human_readable_type %>, <i><%= curation_concern %></i>, changing its visibility to <b><%= visibility_badge(curation_concern.visibility) %></b> until <%= curation_concern.embargo_release_date.to_date.to_formatted_s(:long_ordinal) %>.</p>
-      <p>Would you like to apply the same embargo to all of the files within the  <%= curation_concern.human_readable_type %> as well?</p>
-    <% elsif curation_concern.active_lease? %>
-      <p>You've applied a lease to this <%= curation_concern.human_readable_type %>, <i><%= curation_concern %></i>, changing its visibility to <b><%= visibility_badge(curation_concern.visibility) %></b> until <%= curation_concern.lease_expiration_date.to_date.to_formatted_s(:long_ordinal) %>.</p>
-      <p>Would you like to apply the same lease to all of the files within the  <%= curation_concern.human_readable_type %> as well?</p>
-    <% else %>
       <p>You've changed the permissions on this <%= curation_concern.human_readable_type %>, <i><%= curation_concern %></i>, making it visible to <b><%= visibility_badge(curation_concern.visibility) %></b>.</p>
       <p>Would you like change all of the files within the <%= curation_concern.human_readable_type %> to <b><%= visibility_badge(curation_concern.visibility) %></b> as well?</p>
-    <% end %>
   </div>
   <div class="form-actions panel-footer">
     <%= button_to "Yes please.", copy_curation_concern_permission_path(curation_concern), class: 'btn btn-primary' %>
-    <%= link_to "No. I'll update it manually.", polymorphic_path([:curation_concern, curation_concern]), class: 'btn' %>
+    <%= link_to "No. I'll update it manually.", Sufia::Engine.routes.url_helpers.generic_work_path(curation_concern), class: 'btn' %>
```

Comment:
@scherztc You can pass parameters to I18N and do the ruby code in the parameter.  Here is an example: https://github.com/projecthydra/sufia/blob/pcdm/app/views/generic_files/_browse_everything.html.erb#L3

Although here I might just use

No one—
Notifications

You're receiving notifications because you commented.
2 participants
@scherztc @cam156
Lock pull request
@scherztc
Owner
scherztc commented 15 hours ago

No description provided.
	@scherztc 	Access Rights Should be copied to all the files in the work. Fixes #1001 … 	
98ff564
@cam156 cam156 added the ready for review label 15 hours ago
@cam156 cam156 commented on the diff 4 hours ago
View full changes
...ontrollers/curation_concern/permissions_controller.rb
@@ -0,0 +1,19 @@
+module CurationConcern
+ class PermissionsController < ApplicationController
+ include Worthwhile::CurationConcernController
+ self.curation_concern_type = ActiveFedora::Base
+
+ def confirm
+ end
+
+ def copy_visibility
+ Sufia.queue.push(CopyVisibilityJob.new(params[:id]))
+ flash_message = 'Updating file permissions. This may take a few minutes. You may want to refresh your browser or return to this record later to see the updated file permissions.'
@cam156
Owner
cam156 added a note 4 hours ago

:clap: Nice message. Can we make this I18N and use your text as the default?
@cam156 cam156 commented on the diff 4 hours ago
View full changes
...ontrollers/curation_concern/permissions_controller.rb
@@ -0,0 +1,19 @@
+module CurationConcern
+ class PermissionsController < ApplicationController
+ include Worthwhile::CurationConcernController
+ self.curation_concern_type = ActiveFedora::Base
+
+ def confirm
+ end
+
+ def copy_visibility
+ Sufia.queue.push(CopyVisibilityJob.new(params[:id]))
+ flash_message = 'Updating file permissions. This may take a few minutes. You may want to refresh your browser or return to this record later to see the updated file permissions.'
+ redirect_to Sufia::Engine.routes.url_helpers.generic_work_path(params[:id]), notice: flash_message
@cam156
Owner
cam156 added a note 4 hours ago

@scherztc I wonder if this could be [sufia, curation_concern] Similar to what @hortongn put in for the form.
@scherztc
Owner
scherztc added a note 2 hours ago

Instead of params[:id]?
@cam156
Owner
cam156 added a note 2 minutes ago

@scherztc Instead of Sufia::Engine.routes.url_helpers.generic_work_path(params[:id]) , but here you do not have a curation concern class do you. What you have may be correct...
@cam156 cam156 commented on the diff 4 hours ago
View full changes
app/views/curation_concern/permissions/confirm.html.erb
((14 lines not shown))
</div>
<div class="form-actions panel-footer">
<%= button_to "Yes please.", copy_curation_concern_permission_path(curation_concern), class: 'btn btn-primary' %>
- <%= link_to t(sufia.permission.manual_update"), ... %>